### PR TITLE
Issue#1414 media image

### LINKED ISSUE #1414: Add film + rights tunnel - 10 (upload media)
ref blockframes/storymapping#14
on Zeplin :
https://app.zeplin.io/project/5ca741b3ed21e79faa266664/screen/5e0b5ce0ce5cf519870bb60e

#### location
libs/movie/form/promotional-elements
libs/movie/display/promotional-elements
#### name
MovieFormPromotionalElementsComponent
MovieDisplayPromotionalElementsComponent

- should be refactored from ControlContainer to state
- one image for banner and poster, multiple images for promotional photo

@2amarkina 
image
user can adjust picture size inside of existing cropper
should use the bar?

# E2E Tests to implement 

## Page 10
- [ ] Uploads an image to banner (if possible with cypress)
- [ ] Uploads an image to poster (if possible with cypress)
- [ ] Clicks on "Next" button
- [ ] Redirects to movie tunnel page 11 Files & links

### PR DESCRIPTION
**Pull request by jenna1k**
resolve #1414

### CODE REVIEW

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<tunnel-page>
+  <h4 title>Uploaded Media</h4>
+  <h1 subtitle>Images</h1>
+  <p description class="mat-body-1">Upload your images in .jpeg or .png formats, they will be used for promotional purposes. When you upload an image, we will help you to match the size required.</p>
+  <section tunnel-form>
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html]: do `<ng-container tunnel-form>` Like that you don't have an intermediary tag to deal with

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        <mat-card-title>Movie poster image</mat-card-title>
+      </mat-card-header>
+      <mat-card-content>
+        <ng-container *ngFor="let control of poster.controls">
+          <movie-promotional-elements-images [form]="control" type="poster" [ratio]="5/7" setWidth="500"></movie-promotional-elements-images>
+        </ng-container>
+      </mat-card-content>
+    </mat-card>
+    <!-- Still Photos -->
+    <article *ngFor="let control of stillPhoto.controls; let i = index; let last = last">
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html]: You need to create a mat-card per photo ? 
Also use `ng-container` when using *ngFor to not create more tag than needed.

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      <mat-card>
+        <mat-card-header>
+          <mat-card-title>Still photos</mat-card-title>
+        </mat-card-header>
+        <mat-card-content>
+          <movie-promotional-elements-images [form]="control" type="stillPhoto" [ratio]="7/5" setWidth="500"></movie-promotional-elements-images>
+        </mat-card-content>
+        <!-- Add / Delete -->
+        <footer *ngIf="!last" fxLayout="column" fxLayoutAlign="center center">
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html]: If you really create a card per photo, maybe you can ue the native card-footer then...

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          <button mat-mini-fab color="warn" (click)="stillPhoto.removeAt(i)" aria-label="Remove a director form">
+            <mat-icon>delete</mat-icon>
+          </button>
+          <p>Delete Still</p>
+        </footer>
+        <footer *ngIf="last" fxLayout="column" fxLayoutAlign="center center">
+          <button mat-mini-fab color="primary" (click)="stillPhoto.add()" aria-label="Remove a director form">
+            <mat-icon>add</mat-icon>
+          </button>
+          <p>Add Still</p>
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html]: use span for small text. <p> = paragraph.

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.scss [hunk 5]
```diff
@@ -0,0 +1,11 @@
+mat-card-content {
+  text-align: center;
+}
+
+movie-promotional-elements-images {
+  max-width: 760px;
+}
+
+footer {
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.scss]: I really don't understand why you need that..

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import { Component } from '@angular/core';
+import { MovieForm } from '@blockframes/movie/movie/form/movie.form';
+@Component({
+  selector: 'catalog-tunnel-media-image',
+  templateUrl: './media-image.component.html',
+  styleUrls: ['./media-image.component.scss']
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.ts]: changeDetection

File: libs/movie/src/lib/movie/form/promotional-elements/promotional-elements-images/promotional-elements-images.component.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
+import { MoviePromotionalElementForm } from '../promotional-elements.form';
+@Component({
+  selector: '[form] movie-promotional-elements-images',
+  templateUrl: './promotional-elements-images.component.html',
+  styleUrls: ['./promotional-elements-images.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PromotionalElementsImagesComponent {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/promotional-elements/promotional-elements-images/promotional-elements-images.component.ts]: It feels like you don't need a component for that as you just forward data to the cropper. Just put the cropper in the page and remove this component.

File: libs/ui/src/lib/media/cropper/cropper/cropper.component.html [hunk 8]
```diff
@@ -4,14 +4,16 @@
...
+    <ng-content select="[dimensions]"></ng-content>
+    <p class="mat-caption">Drag and drop a file here or <a color="primary" routerLink="#" (click)="fileUploader.click()">Browse</a></p>
   </article>
   <article class="cropper" *ngSwitchCase="'crop'" @zoomDelay>
     <image-cropper
       [imageFile]="file"
       [aspectRatio]="cropRatio"
       [containWithinAspectRatio]="true"
-      [resizeToWidth]="parentWidth"
+      [resizeToWidth]="setWidth? setWidth : parentWidth"
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/media/cropper/cropper/cropper.component.html]: you can do `setWidth || parentWidth` (it only works if setWidth shouldn't be 0)

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.scss (hunk 5 above)
Comment (jenna1k) [reply apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.scss]: removed

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html (hunk 2 above)
Comment (jenna1k) [reply apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html]: moved inside mat-card-content

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html (hunk 1 above)
Comment (jenna1k) [reply apps/catalog/catalog/src/app/dashboard/movie-tunnel/media-image/media-image.component.html]: I got a weird effect that only first card takes full width and others shrink when I change it to ng-container

File: libs/movie/src/lib/movie/form/promotional-elements/promotional-elements-images/promotional-elements-images.component.ts (hunk 7 above)
Comment (jenna1k) [reply libs/movie/src/lib/movie/form/promotional-elements/promotional-elements-images/promotional-elements-images.component.ts]: removed

**Pull request merged**